lms/djangoapps/grades/management/commands/tests/test_reset_grades.py

```diff
@@ -0,0 +1,289 @@
+
+        self._assert_grades_absent_for_courses(self.course_keys)
+        self._assert_subsection_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * self.num_courses * self.num_subsections,
+        )
+        self._assert_course_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * self.num_courses,
+        )
+
+    @patch('lms.djangoapps.grades.management.commands.reset_grades.log')
+    @ddt.data(1, 2, 3)
+    def test_reset_some_courses(self, num_courses_to_reset, mock_log):
+        self._update_or_create_grades()
+        self._assert_grades_exist_for_courses(self.course_keys)
+
+        self.command.handle(
+            delete=True,
+            courses=[unicode(course_key) for course_key in self.course_keys[:num_courses_to_reset]]
+        )
+
+        self._assert_grades_absent_for_courses(self.course_keys[:num_courses_to_reset])
+        self._assert_grades_exist_for_courses(self.course_keys[num_courses_to_reset:])
+        self._assert_subsection_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * num_courses_to_reset * self.num_subsections,
+        )
+        self._assert_course_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * num_courses_to_reset,
+        )
+
+    def test_reset_by_modified_start_date(self):
+        self._update_or_create_grades()
+        self._assert_grades_exist_for_courses(self.course_keys)
+
+        num_courses_with_updated_grades = 2
+        with freeze_time(self._date_from_now(days=4)):
+            self._update_or_create_grades(self.course_keys[:num_courses_with_updated_grades])
+
+        self.command.handle(delete=True, modified_start=self._date_str_from_now(days=2), all_courses=True)
+
+        self._assert_grades_absent_for_courses(self.course_keys[:num_courses_with_updated_grades])
+        self._assert_grades_exist_for_courses(self.course_keys[num_courses_with_updated_grades:])
+
+    def test_reset_by_modified_start_end_date(self):
+        self._update_or_create_grades()
+        self._assert_grades_exist_for_courses(self.course_keys)
+
+        with freeze_time(self._date_from_now(days=3)):
+            self._update_or_create_grades(self.course_keys[:2])
+        with freeze_time(self._date_from_now(days=5)):
+            self._update_or_create_grades(self.course_keys[2:4])
+
+        self.command.handle(
+            delete=True,
+            modified_start=self._date_str_from_now(days=2),
+            modified_end=self._date_str_from_now(days=4),
+            all_courses=True,
+        )
+
+        # Only grades for courses modified within the 2->4 days
+        # should be deleted.
+        self._assert_grades_absent_for_courses(self.course_keys[:2])
+        self._assert_grades_exist_for_courses(self.course_keys[2:])
+
+    @patch('lms.djangoapps.grades.management.commands.reset_grades.log')
+    def test_dry_run_all_courses(self, mock_log):
```

Comment:
Mostly meant that it wasn't the fastest query (took a few seconds).  May have been cold cache, may be something deeper I'm misunderstanding.  I'm not too worried about it.